Introduce Rake and Minitest, and write tests

kozy4324 opened the pull request:
Thank you for this awesome add-on! I’ve been wanting to create a Brakeman add-on.

I wrote tests for ruby-lsp-brakeman based on the tests for ruby-lsp and ruby-lsp-rails. I chose Minitest as the testing framework, but there's no strong preference for it.

Additionally, I addressed the following two points:

- Changed the behavior to fall back with force_scan: true when a Brakeman::NoApplication exception occurs during Brakeman.run execution.
- Modified the deactivate callback to terminate child threads.

The fallback with `force_scan: true` might be too forceful for test execution purposes. Please let me know if there's a better approach.

Thanks again for your excellent work on this add-on. Looking forward to any feedback or suggestions!

presidentbeef commented:
Hey @kozy4324 - apologies, I don't remember seeing this PR come through!

> The fallback with force_scan: true might be too forceful for test execution purposes. 

I don't think people want Brakeman to run on non-Rails projects :thinking: one option would be to also generate a temporary file (`config/brakeman.yml`) with a Brakeman configuration that sets `force_scan`:

```yaml
---
force_scan: true
```

kozy4324 commented:
Hi @presidentbeef,
Thank you for your advice!

I updated the tests to create the config file in setup and teardown.
Although creating it every time is a bit redundant, since the test count is still small, I opted for this simpler implementation for now.

If the config file does not exist, it will be created temporarily; if it exists, it will be used. However, if it does not include force_scan: true, the test will fail with a message indicating this.

Additionally, since running tests with ruby-lsp 0.23.12 or later causes a stack infinite loop at `server.pop_response until ...`, I also fixed the test helper methods accordingly.
The related ruby-lsp PR is https://github.com/Shopify/ruby-lsp/pull/3269, and those changes are included here as well.

Thanks again for your support!

presidentbeef commented:
Thank you!
